lib/marloss/locker.rb

```diff
@@ -22,9 +22,12 @@ def release_lock
       store.delete_lock(name)
     end
 
-    def wait_until_lock_obtained(sleep_seconds: 3)
+    def wait_until_lock_obtained(sleep_seconds: 3, retries: 3)
```

Comment:
I do agree with the fact that using a spinlock is generally not great but some people may rely on it.
As you said this would be a breaking change and I'd be better to make this change less disruptive.
just out of curiosity, how did you pick that number `3`?
I'm thinking that it could be set to `20`, with the current 3s sleep it'd wait for ~1min making it a bit less disruptive change.
Otherwise, setting the retries to `nil` and skip this new logic if retries is not set, could also be a nicer way to introduce this.
would be great to know the reasoning and use cases behind this change

Reply:
Three is just the standard for the codebase we are working in. In a high concurrency env we want to just move on to the next work unit if we are unable to obtain a lock.

Perhaps a better way to do this is to add a new method for `obtain_lock` and keep `wait_until_lock_obtained` as a spin lock.

Reply:
the base method that creates the lock is called `obtain_lock`
https://github.com/eredi93/marloss/blob/d0a87d41ac0613c4b7a4aa18fe707f3ec21994ff/lib/marloss/locker.rb#L13

what about creating a new method called `obtain_lock_with_retries`?